system/cle/cle.c

```diff
@@ -1124,11 +1149,20 @@ int cle_fd(FAR char *line, FAR const char *prompt, uint16_t linelen,
 
   /* Get the current cursor position */
 
-  ret = cle_getcursor(&priv, &priv.row, &column);
-
-  if (ret < 0)
+  for (; ; )
     {
-      return ret;
+      ret = cle_getcursor(&priv, &priv.row, &column);
```

Comment:
but why the error doesn't happen before ECHO change?

Reply:
In fact this issue is not due to ECHO but https://github.com/apache/nuttx-apps/commit/95f32fd018be7a66354942647d9e1c778ad5d488 , 
seems c3's usb-serial driver need more time to get ready. detail please refer to https://github.com/apache/nuttx/issues/8731#issuecomment-1461288702

Reply:
should we fix the usb serial driver instead

Reply:
I think usb driver should be fixed, but this patch is also needed since nsh shold not hang here if can't read vt100 command respons.

Reply:
but the response is enter by user, you can't expect how fast the user will generate the input.

Reply:
No, the response here is sended by terminal auto.

Reply:
but user may not open terminal yet